imports/plugins/core/email/client/containers/emailConfig.js

```diff
@@ -33,15 +33,33 @@ const wrapComponent = (Comp) => (
     }
 
     componentWillMount() {
+      this.checkEmailStatus();
+    }
+
+    componentWillReceiveProps(nextProps) {
+      const { settings } = this.props;
+      const { settings: nextSettings } = nextProps;
+      // if the email settings do not match check the email status
+      if (JSON.stringify(settings) !== JSON.stringify(nextSettings)) {
+        this.checkEmailStatus();
+      } else {
+        return;
+      }
+    }
+
+    // checking email settings
+    // and updating status
+    checkEmailStatus() {
       const { settings } = this.props;
       const { service, host, port, user, password } = settings;
 
       if (service && host && port && user && password) {
         Meteor.call("email/verifySettings", (error) => {
           if (error) {
             this.setState({ status: "error" });
+          } else {
+            this.setState({ status: "valid" });
           }
-          this.setState({ status: "valid" });
```

Comment:
Anytime calling `setState` in a callback, we should get in the habit of checking to make sure the component is still mounted, to avoid errors in the console from React. The simplest way is to add componentDidMount and componentWillUmount, setting `this._isMouted` to true/false in those. And then in this callback, `if (!this._isMounted) return;` at the top.